sympy/solvers/ode.py

```diff
@@ -1482,6 +1491,63 @@ def _test_term(coeff, order):
     else:
         return tuple(retlist)
 
+def ode_2nd_hypergeometirc(eq, func, order, match):
+
+    x = func.args[0]
+    C0, C1 = get_numbered_constants(eq, num=2)
+    a = match['a']
+    b = match['b']
+    c = match['c']
+    x0 = match['x0']
+
+    if x0 == 0:
+        if c.is_integer == False:
+            return Eq(func, C0*hyper([a, b], [c], x) + C1*hyper([a-c+1, b-c+1], [2-c], x)*x**(1-c))
+
+    raise NotImplementedError(" .....")
+
+def match_2nd_hypergeometric(r, func, x0):
+
+    from sympy.polys.polytools import factor
+    # eq = a3*f(x).diff(x, 2) + b3*f(x).diff(x) + c3*f(x)
+    f = func
+    x = func.args[0]
+    df = f.diff(x)
+    a = Wild('a', exclude=[x,f,df])
+    b = Wild('b', exclude=[x,f,df])
+    c = Wild('c', exclude=[x,f,df])
+    d = Wild('d', exclude=[x,f,df])
+    a3 = Wild('a3', exclude=[f, df, f.diff(x, 2)])
+    b3 = Wild('b3', exclude=[f, df, f.diff(x, 2)])
+    c3 = Wild('c3', exclude=[f, df, f.diff(x, 2)])
+    if r[c3].has(x):
+        coeff3 = factor(r[c3]).match(a*(x-b)**c)
+        if coeff3:
+            r[a3] = cancel(r[a3]/((x-coeff3[b])**(coeff3[c])))
+            r[b3] = cancel(r[b3]/((x-coeff3[b])**(coeff3[c])))
+            r[c3] = cancel(r[c3]/((x-coeff3[b])**(coeff3[c])))
+        else:
+            return None
+    coeff1 = (r[a3]).match(a*x-a*x**2)
+    if coeff1:
+        if coeff1[a]!=1:
```

Comment:
What about `if coeff1 and coeff[a] != 1`? May be, `coeff[a]!=1` -> `coeff[a] != 1`(spaces for readability).

Reply:
If `coeff1` is a `dict`, then probably, `coeff1.get(a, None) != None` should work.